xkomhotshot

```diff
@@ -31,23 +32,40 @@ def get_hot_shot():
     return hot_shot_info
 
 
+def build_html_caption(data: dict) -> str:
+    name = escape(data["name"])
+    price_promo = data["price_promo"]
+    price_regular = data["price_regular"]
+    # Safeguard against division by zero
+    if price_regular and price_regular != 0:
+        discount = round((price_regular - price_promo) / price_regular * 100)
+    else:
+        discount = 0
+    price_min = data["price_min"]
+    price_min_date = data["price_min_date"]
+    return (
+        f"<b>{name}</b>\n"
+        f"Cena promocyjna: <b>{price_promo} PLN (-{discount}%)</b>\n"
+        f"Cena regularna: <b>{price_regular} PLN</b>\n"
+        f"Cena najniższa z ostatnich 30 dni: <b>{price_min} PLN ({price_min_date})</b>\n"
+        f"Ilość sztuk: <b>{data['item_count']}</b>\n"
+        f"Link: {data['link']}"
```

Comment:
oh, and the link should be escaped too

Reply:
maybe the URL construction should be moved from `get_hot_shot` to `build_html_caption`, so that only the ID has to be escaped